dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3753,12 +3753,14 @@ DDS::ReturnCode_t
 Sedp::write_dcps_participant_secure(const Security::SPDPdiscoveredParticipantData& msg,
                                     const RepoId& part)
 {
-  static DCPS::SequenceNumber sequence = 0;
-
   DCPS::RepoId remote_reader(part);
-  remote_reader.entityId = ENTITYID_SPDP_RELIABLE_BUILTIN_PARTICIPANT_SECURE_READER;
+  if (part != GUID_UNKNOWN) {
+    remote_reader.entityId = ENTITYID_SPDP_RELIABLE_BUILTIN_PARTICIPANT_SECURE_READER;
+    return dcps_participant_secure_writer_->write_dcps_participant_secure(msg, remote_reader, seq_++);
+  }
 
-  return dcps_participant_secure_writer_->write_dcps_participant_secure(msg, remote_reader, ++sequence);
+  DCPS::SequenceNumber sequence = DCPS::SequenceNumber::SEQUENCENUMBER_UNKNOWN();
+  return dcps_participant_secure_writer_->write_dcps_participant_secure(msg, remote_reader, sequence);
```

Comment:
Use seq_.  Also rename it so we know what it does (`participant_secure_sequence_`)